services/src/modules/apollo-server-plugins/introspection-query-policy.ts

```diff
@@ -0,0 +1,47 @@
+import { GraphQLResolveInfo, SchemaMetaFieldDef } from 'graphql';
+import { ForbiddenError } from 'apollo-server-core';
+import logger from '../logger';
+import { RequestContext } from '../context';
+import { ResourceGroup } from '../resource-repository';
+import { ActiveDirectoryAuth } from '../upstreams/authentication';
+import { PolicyExecutor } from '../directives/policy';
+
+const INTROSPECTION_QUERY_FIELD_NAMES = new Set(['__schema', '__type']);
+
+function isIntrospectionRequest(info: GraphQLResolveInfo) {
+  return INTROSPECTION_QUERY_FIELD_NAMES.has(info.fieldName);
+}
+
+export function createIntrospectionQueryPolicyPlugin(): void {
+  const originalSchemaMetaFieldResolver = SchemaMetaFieldDef.resolve;
+
+  SchemaMetaFieldDef.resolve = function (source, args, context, info) {
+    const { resourceGroup, policyExecutor } = (context as unknown) as RequestContext;
```

Comment:
Apollo server performs introspection query request w/o sending context [here](https://github.com/apollographql/apollo-server/blob/94862a48ff9430d3238f9b456c9a99159fd0aa69/packages/apollo-server-core/src/utils/schemaHash.ts#L12). So the context will be undefined and destructuring will fail. 
So in the case the context is null we need to skip policy check.
I think it's the cause of the infinitive schema update issue you've experienced.